README-FRONT-END-DEV.md

```diff
@@ -0,0 +1,204 @@
+
+This uses the `POST /projects` functionality of the Messenger service, and should only be done **once.** Attempting to register the same project multiple times will net you a bunch of errors.
+
+Here's an example of registering (on `staging`) Project 1651, aka `darkeshard/transfromers`:
+
+```
+import apiClient from 'panoptes-client/lib/api-client.js';
+
+var url = 'https://messenger-staging.zooniverse.org/projects/' +
+          '?slug=' + encodeURIComponent('darkeshard/transformers');  //OK, honestly, I'm not sure if this ?slug part is necessary, but my personal notes says yes? (@shaun.a.noordin 20170630)
+
+var body = JSON.stringify({  //Reminder: turn your JSON objects into strings
+  'data': {
+    'attributes': {
+      'slug': 'darkeshard/transformers'
+    }
+  }
+});
+
+const opt = {
+  method: 'POST',
+  mode: 'cors',
+  headers: new Headers({
+    'Authorization': apiClient.headers.Authorization,
+    'Content-Type': 'application/json',
+  }),
+  body: body,
+};
+
+console.log('REGISTER PROJECT: start');
+
+fetch(url, opt)
+.then((response) => {
+  if (response.status < 200 || response.status > 202) { return null; }
+  return response.json();
+})
+.then((json) => {
+  if (json && json.data) {
+    console.log('REGISTER PROJECT: completed.');
+  }
+})
+.catch((err) => {
+  ...
+});
+
+```
+
+Expected Results:
+* If successful, you should see the `darkeshard/transformers` Project listed when you access `GET /projects/1651`.
+
+Common Problems:
+* If you're seeing 401 Unauthorized errors, chances are you're either:
+  * Not logged into Panoptes.
+  * Logged into `staging` instead of `production` or vice versa.
+  * Logged into an account that's not the owner/collaborator of the Project.
+* Strangely, we've also experienced 401 errors when the `body` string isn't properly structured. Double check your `body`: it should just be data->attributes->slug.
+
+## 2. Add a Transcription
+
+If a Transcription does not yet exist in the Messenger's database, you'll first need to add it using `POST /transcriptions`.
+
+Here's an example of adding a Transcription for (staging) Subject 48407, which is a picture of Optimus Prime.
+
+```
+import apiClient from 'panoptes-client/lib/api-client.js';
+
+var url = 'https://messenger-staging.zooniverse.org/transcriptions/';
+
+var body = JSON.stringify({
+  'data': {
+    'attributes': {
+      'id': '48407',
+      'text': 'Hello Optimus Prime',
+      'status': 'amended',
+    },
+    'relationships': {
+      'project': {
+        'data': {
+          'type': 'projects',
+          'id': '1651',
+        }
+      }
+    },
+  }
+});
+
+const opt = {
+  method: 'POST',
+  mode: 'cors',
+  headers: new Headers({
+    'Authorization': apiClient.headers.Authorization,
+    'Content-Type': 'application/json',
+  }),
+  body: body,
+};
+
+fetch(url, opt)
+.then((response) => {
+  ...
+})
+```
+
+There's also an alternative way to structure the body, but maybe don't do this?
```

Comment:
This also seems confusing, I'd rather we raise an issue and figure out what's up.